src/python/pants/engine/internals/parser_test.py

```diff
@@ -122,6 +125,46 @@ def perform_test(extra_targets: list[str], dym: str) -> None:
     perform_test(test_targs, dym_many)
 
 
+def test_unrecognized_symbol_during_bootstrap_issue_19156(
+    defaults_parser_state: BuildFileDefaultsParserState,
+) -> None:
+    build_file_aliases = BuildFileAliases(
+        objects={"obj": 0},
+        context_aware_object_factories={"caof": lambda parse_context: lambda _: None},
+    )
+    parser = Parser(
+        build_root="",
+        registered_target_types=RegisteredTargetTypes({"tgt": GenericTarget}),
+        union_membership=UnionMembership({}),
+        object_aliases=build_file_aliases,
+        ignore_unrecognized_symbols=True,
+    )
+    prelude_symbols = BuildFilePreludeSymbols.create({"prelude": 0}, ())
+    target_adaptors = parser.parse(
+        "dir/BUILD",
+        "tgt(field=fake(42,a=(), b='c'))",
+        prelude_symbols,
+        EnvironmentVars({}),
+        False,
+        defaults_parser_state,
+        dependents_rules=None,
+        dependencies_rules=None,
+    )
+
+    assert len(target_adaptors) == 1
+    raw_field = target_adaptors[0].kwargs["field"]
+    assert repr(raw_field) == "fake(42, a=(), b='c')"
+
+    class TestField(StringField):
+        alias = "field"
+
+    err = re.escape(
+        f"The 'field' field in target // must be a string, but was `{raw_field!r}` with type `_UnrecognizedSymbol`."
```

Comment:
That type seems dubious. Why are we surfacing the name of an internal type?

Reply:
The error might need to change to detect `_UnrecognizedSymbol` so that this error reads:
```suggestion
        f"The 'field' field in target // must be a string, but was `{raw_field!r}` which is an unrecognized symbol."
```

Reply:
Good point, as it may surface to the user like this, it ought maybe not be marked internal like that; or perhaps better adjust the error message for this particular case. I'll take another swing at it.